### PR TITLE
bug-fix: No video thumbnail on quote reply

### PR DESCRIPTION
**Pull request by liamcho**
Fixes: [UIKIT-4525](https://sendbird.atlassian.net/browse/UIKIT-4525

Screen Shot 2023-10-17 at 2 46 42 PM
Screen Shot 2023-10-17 at 2 47 10 PM
)

[UIKIT-4525]: https://sendbird.atlassian.net/browse/UIKIT-4525?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/ui/QuoteMessageInput/QuoteMessageThumbnail.tsx [hunk 1]
```diff
@@ -8,11 +8,12 @@ import {
   isFileMessage, isImageFileInfo,
   isImageMessage,
   isMultipleFilesMessage,
-  isThumbnailMessage,
+  isThumbnailMessage, isVideoMessage,
```
Comment (AhyoungRyu) [inline src/ui/QuoteMessageInput/QuoteMessageThumbnail.tsx:11]: nit

```suggestion
  isThumbnailMessage,
  isVideoMessage,
```

File: src/ui/QuoteMessageInput/QuoteMessageThumbnail.tsx [hunk 2]
```diff
@@ -40,6 +59,18 @@ export default function QuoteMessageThumbnail({ message }: Props): ReactElement
         fixedSize
       />
     );
+  } else if (isVideoMessage(message as FileMessage) && thumbnailUrl) {
```
Comment (AhyoungRyu) [inline src/ui/QuoteMessageInput/QuoteMessageThumbnail.tsx:62]:
```
if (isVideoMessage(message as FileMessage) && thumbnailUrl) {
   ...
} else if (isThumbnailMessage(message) || isMultipleFilesMessage(message)) {
   ...
}
```

I think we can make a bit simpler here by changing the order of these two.

**Review by AhyoungRyu — APPROVED** [review]

File: src/ui/QuoteMessageInput/QuoteMessageThumbnail.tsx [hunk 3]
```diff
@@ -24,12 +25,30 @@ export default function QuoteMessageThumbnail({ message }: Props): ReactElement
...
+        || (
+          castedMessage.fileInfoList.length > 0
+          && isImageFileInfo((message as MultipleFilesMessage).fileInfoList[0])
+          && getMessageFirstFileUrl(message)
+        );
+    })
+    .otherwise(() => {
+      return getMessageFirstFileThumbnailUrl(message);
+    });
+  if (!isVideoMessage(message) && (isThumbnailMessage(message) || isMultipleFilesMessage(message)) && thumbnailUrl) {
```
Comment (bang9) [inline src/ui/QuoteMessageInput/QuoteMessageThumbnail.tsx:51]: How about this?

```tsx
  let thumbnailUrl = getMessageFirstFileThumbnailUrl(message);
  if (!thumbnailUrl && message.isFileMessage() && isImageMessage(message)) {
    thumbnailUrl = getMessageFirstFileUrl(message);
  }
  if (!thumbnailUrl && message.isMultipleFilesMessage() && isImage(message.fileInfoList?.[0].mimeType)) {
    thumbnailUrl = message.fileInfoList[0].url;
  }
  
  if (thumbnailUrl) {
    return (
      <ImageRenderer
        className={componentClassname}
        url={thumbnailUrl}
        alt={getMessageFirstFileType(message)}
        width="44px"
        height="44px"
        fixedSize
      />
    );
  }
```

Comment (liamcho) [reply src/ui/QuoteMessageInput/QuoteMessageThumbnail.tsx:52]: better 👍

File: src/ui/QuoteMessageInput/QuoteMessageThumbnail.tsx (hunk 2 above)
Comment (liamcho) [reply src/ui/QuoteMessageInput/QuoteMessageThumbnail.tsx:63]: nice!

File: src/ui/QuoteMessageInput/QuoteMessageThumbnail.tsx (hunk 3 above)
Comment (liamcho) [reply src/ui/QuoteMessageInput/QuoteMessageThumbnail.tsx:52]: how come copying the code snippet makes paste into a single line? 😢

File: src/ui/QuoteMessageInput/QuoteMessageThumbnail.tsx [hunk 4]
```diff
@@ -24,12 +25,51 @@ export default function QuoteMessageThumbnail({ message }: Props): ReactElement
...
+        className={componentClassname}
+        url={thumbnailUrl}
+        alt={getMessageFirstFileType(message)}
+        width="44px"
+        height="44px"
+        fixedSize
+      />
+    );
+  }
+  if (isVideoMessage(message as FileMessage) && thumbnailUrl) {
```
Comment (bang9) [inline src/ui/QuoteMessageInput/QuoteMessageThumbnail.tsx:60]: it seems not reachable if `thumbnailUrl` is available.

File: src/ui/QuoteMessageInput/QuoteMessageThumbnail.tsx [hunk 5]
```diff
@@ -24,12 +25,51 @@ export default function QuoteMessageThumbnail({ message }: Props): ReactElement
...
-      && getMessageFirstFileUrl(message));
-  if ((isThumbnailMessage(message) || isMultipleFilesMessage(message)) && thumbnailUrl) {
+  let thumbnailUrl = getMessageFirstFileThumbnailUrl(message);
+  if (!thumbnailUrl) {
+    if (
+      message.isFileMessage?.()
+      && (
+        isImageMessage(message as FileMessage)
+        || isVideoMessage(message as FileMessage)
+      )
```
Comment (bang9) [inline src/ui/QuoteMessageInput/QuoteMessageThumbnail.tsx:35]:
```suggestion
      message.isFileMessage()
      && (isImageMessage(message) || isVideoMessage(message)
      )
```
Type inference can be done by using message.isFileMessage().

Comment (liamcho) [reply src/ui/QuoteMessageInput/QuoteMessageThumbnail.tsx:35]: So I get isFileMessage is undefined error sometimes when app is running and test also is failing. Idk why but I started avoid using message.isXxxMessage(). Instead I use isXxxMessage(message).

File: src/ui/QuoteMessageInput/QuoteMessageThumbnail.tsx (hunk 4 above)
Comment (liamcho) [reply src/ui/QuoteMessageInput/QuoteMessageThumbnail.tsx:60]: Thanks for the catch. It is a duplicate

File: src/ui/QuoteMessageInput/QuoteMessageThumbnail.tsx (hunk 5 above)
Comment (bang9) [reply src/ui/QuoteMessageInput/QuoteMessageThumbnail.tsx]: In the test or storybook?

Comment (liamcho) [reply src/ui/QuoteMessageInput/QuoteMessageThumbnail.tsx]: yes in storybook. Also when running test. Yes we need to find cause and fix it.

File: src/ui/QuoteMessageInput/QuoteMessageThumbnail.tsx [hunk 6]
```diff
@@ -30,33 +30,21 @@ export default function QuoteMessageThumbnail({ message }: Props): ReactElement
...
+        || isVideoMessage(message)
       )
     ) {
       thumbnailUrl = getMessageFirstFileUrl(message);
     } else if (
       message.isMultipleFilesMessage?.()
       && (
         message.fileInfoList.length > 0
-        && isImageFileInfo((message as MultipleFilesMessage).fileInfoList[0])
+        && isImageFileInfo((message).fileInfoList[0])
```
Comment (bang9) [inline src/ui/QuoteMessageInput/QuoteMessageThumbnail.tsx:42]: need to check isVideoFileInfo maybe 🌝?

Comment (liamcho) [reply src/ui/QuoteMessageInput/QuoteMessageThumbnail.tsx:42]: Ah no but good similar. MFM supports image including GIF. Video is not supported.

**Review by bang9 — APPROVED** [review]

**Pull request merged**